src/frontend/screens/ObservationCreate/index.tsx

```diff
@@ -169,12 +210,56 @@ export const ObservationCreate = ({
     value,
   ]);
 
+  const checkAccuracyAndLocation = React.useCallback(() => {
+    const confirmationOptions: AlertButton[] = [
+      {
+        text: formatMessage(m.saveAnyway),
+        onPress: createObservation,
+        style: 'default',
+      },
+      {
+        text: formatMessage(m.manualEntry),
+        onPress: () => navigation.navigate('ManualGpsScreen'),
+        style: 'cancel',
+      },
+      {
+        text: formatMessage(m.keepWaiting),
+        onPress: () => {},
+      },
+    ];
+
+    if (!value) throw new Error('no observation saved in persisted state ');
+
+    const accuracy = value.metadata?.position?.coords?.accuracy;
+
+    if (!liveLocation) {
+      Alert.alert(
+        formatMessage(m.noGpsTitle),
+        formatMessage(m.noGpsDesc),
+        confirmationOptions,
+      );
+      return;
+    }
+
+    // If we don't have accuracy, allow save anyway (this is a remnant from mapeo: https://github.com/digidem/mapeo-mobile/blob/0c0ebbb9ef2261e21cd1d1c8bd5ab2fe42017ea3/src/frontend/screens/ObservationEdit/SaveButton.js#L125C3-L125C50)
+    if (accuracy && !isGpsAccurate(accuracy)) {
+      Alert.alert(
+        formatMessage(m.weakGpsTitle),
+        formatMessage(m.weakGpsDesc),
+        confirmationOptions,
+      );
+      return;
+    }
```

Comment:
Ah with the fix applied in 5aab158e01d7a7c9f1fad97c5c22da0b37c80e78 i think that the situation I wrote above doesn't happen anymore. Still think it might be worth having that check anyways for clarity though. The `liveLocation` variable is a little confusing to me at a glance otherwise

Reply:
https://github.com/digidem/comapeo-mobile/pull/702/commits/d3519214756fc2fbe99980e4ad1e4ca8fb6024f9